packages/@ionic/cli-utils/src/lib/project/angular/build.ts

```diff
@@ -18,73 +44,45 @@ export class BuildRunner extends BaseBuildRunner<AngularBuildOptions> {
       description: `
 ${chalk.green('ionic build')} uses the Angular CLI. Use ${chalk.green('ng build --help')} to list all Angular CLI options for building your app. See the ${chalk.green('ng build')} docs${chalk.cyan('[1]')} for explanations. Options not listed below are considered advanced and can be passed to the ${chalk.green('ng')} CLI using the ${chalk.green('--')} separator after the Ionic CLI arguments. See the examples.
 
-${chalk.cyan('[1]')}: ${chalk.bold('https://github.com/angular/angular-cli/wiki/build#ng-build')}`,
-      options: [
-        {
-          name: 'dev',
-          summary: `Sets the build target to ${chalk.green('development')}`,
-          type: Boolean,
-          hint: 'ng',
-        },
-        {
-          name: 'prod',
-          summary: `Sets the build target to ${chalk.green('production')}`,
-          type: Boolean,
-          hint: 'ng',
-        },
-        {
-          name: 'target',
-          summary: 'Set the build target to a custom value',
-          aliases: ['t'],
-          groups: [OptionGroup.Advanced],
-          hint: 'ng',
-        },
-        {
-          name: 'environment',
-          summary: 'Set the build environment to a custom value',
-          aliases: ['e'],
-          groups: [OptionGroup.Advanced],
-          hint: 'ng',
-        },
-      ],
+${chalk.cyan('[1]')}: ${chalk.bold('https://github.com/angular/angular-cli/wiki/build')}`,
+      options: NG_BUILD_OPTIONS,
     };
   }
 
   createOptionsFromCommandLine(inputs: CommandLineInputs, options: CommandLineOptions): AngularBuildOptions {
     const baseOptions = super.createBaseOptionsFromCommandLine(inputs, options);
-    let target = options['target'] ? String(options['target']) : undefined;
-    const environment = options['environment'] ? String(options['environment']) : undefined;
-
-    if (!target) {
-      if (options['dev']) {
-        target = 'development';
-      } else if (options['prod']) {
-        target = 'production';
-      }
-    }
+    const ngOptions = NG_BUILD_OPTIONS
+      .filter(option => options[option.name] !== null && options[option.name] !== undefined)
+      .reduce((accum, option) => {
+        accum[option.name] = options[option.name];
+
+        return accum;
+      }, {} as any);
 
     return {
       ...baseOptions,
+      ...ngOptions,
       type: 'angular',
-      target,
-      environment,
     };
   }
 
   async buildOptionsToNgArgs(options: AngularBuildOptions): Promise<string[]> {
+
+    const ngOptions = NG_BUILD_OPTIONS
+      .map(option => option.name as keyof AngularBuildOptions)
+      .filter(option => options[option] !== null && options[option] !== undefined)
+      .reduce((accum, option) => {
+        accum[option] = String(options[option]);
+
+        return accum;
+      }, {} as any);
+
     const args: ParsedArgs = {
       _: [],
-      environment: options.environment,
+      ...ngOptions,
+      'output-path': 'www',
```

Comment:
I believe this is specified in angular.json. No reason to specify it here

Reply:
I was thinking for a CI/CD scenario it might be useful to be able to run `ng build` separately which could build each app to a separate folder (as configured in angular.json), whereas running `ionic cordova build` should always build to `www` so cordova can pick it up.

The alternative here is to require the output path for every app in angular.json to point at `www`, or copy folders from the output path into `www` during the build process.

Reply:
Ohh, okay. Please check if `options.engine === 'cordova'` then. This logic is reused for `ionic build` as well as `ionic cordova build`